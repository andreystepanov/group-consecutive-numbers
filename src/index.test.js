import group from './'

test('defined', () => {
  expect(group).toBeDefined()
  expect(typeof group).toBe('function')
})

test('removes duplicates and sorts numbers', () => {
  const result = group([1, 2, 4, 6, 8, 11, 7])
  expect(result).toMatchSnapshot()
})

test('ignores all non number values', () => {
  const result = group([1, 2, 4, 'notnumber', 8, 11, 7])
  expect(result).toMatchSnapshot()
})

test('return only groups with min length', () => {
  const result = group([1, 2, 4, 6, 9, 8, 11, 7], 3)
  expect(result).toMatchSnapshot()
})

test.skip('handles float numbers correctly', () => {
  // const result = group([1, 2, 4, 6.2, 6.1, 6.3, 8, 11, 7])
  // expect(result).toMatchSnapshot()
})
