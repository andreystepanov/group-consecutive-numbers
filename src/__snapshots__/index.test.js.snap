// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignores all non number values 1`] = `
Array [
  Array [
    1,
    2,
  ],
  Array [
    7,
    8,
  ],
]
`;

exports[`removes duplicates and sorts numbers 1`] = `
Array [
  Array [
    1,
    2,
  ],
  Array [
    6,
    7,
    8,
  ],
]
`;

exports[`return only groups with min length 1`] = `
Array [
  Array [
    6,
    7,
    8,
    9,
  ],
]
`;
